### ISSUE TITLE
`dotnet new react ...`  generates a JavaScript React project instead of TypeScript after updating to 2.1.300 (.NET Core 2.1)

### ISSUE THREAD

**Issue by livarcocc** [aspnet/templating#572]
_From @Nufflee on June 14, 2018 12:49_

I'm opening this issue because that command would generate a TypeScript project in SDK version 2.1.200 (.NET Core version 2.0.8). In the changelog it states that you've switched over to create-react-app but it, by default, generates a JavaScript React project. I'm not sure if this change was intended as I couldn't find anything about that in the change log, but if so, you should probably provide an option to switch to TypeScript (you can use [create-react-app-typescript](https://github.com/wmonk/create-react-app-typescript) to generate a typescript project). 

## Steps to reproduce
Run `dotnet new -o MyProject` and observe that code in ClientApp directory is JavaScript not TypeScript.

## Expected  behavior
Code in ClientApp directory should be TypeScript.

## Actual behavior
Code ub ClientApp directory is JavaScript.

## Environment data
`dotnet --info` output:
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300\

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.200 [C:\Program Files\dotnet\sdk]
  2.1.300 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

_Copied from original issue: dotnet/cli#9489_

**Comment by tpetrina** [comment]
I think this is pretty much intended.

**Comment by Nufflee** [comment]
How so? It was generating a TypeScript project before, there's nothing in the changelog about this change nor there's a way to toggle between JavaScipt and TypeScript. @tpetrina

**Comment by tpetrina** [comment]
See:

 - https://github.com/aspnet/templating/issues/242#issuecomment-359865442.
 - https://github.com/aspnet/JavaScriptServices/issues/997#issuecomment-344603801

**Comment by Nufflee** [comment]
Okay, I see. But I still don't get @SteveSandersonMS' exactly, I see it's bad to generate JS apps by default as most tutorials are written for it, but why not make a switch for people that know what they're doing?

**Comment by tpetrina** [comment]
Because then they have to support it.

**Comment by Meligy** [comment]
That's a sad change. I'm working on a big project with tens of developers in a couple of countries that started with the .NET Core React template when it supported TypeScript.

**Comment by Meligy** [comment]
Can we re-open this as a feature request please?

Thanks for awesome templates.

**Comment by Meligy** [comment]
Would it be as simple as adding ` --scripts-version=react-scripts-ts` to the command calling `create-react-app`?

As in https://github.com/wmonk/create-react-app-typescript

**Comment by tpetrina** [comment]
Yes you can, I do that. However, if Microsoft ships official template with that, where would you report bugs, features or any feedback? Over at wmonk's repo or here? That is what is meant by support.

Also, it is better to use `react-app-rewired` instead.

**Comment by jtheisen** [comment]
@tpetrina Well, they damn well should support that. It's the only technological combination that lets you wire the ui to the business model in a type safe manner. It's state-of-the-art.

I agree that sometimes it's overkill: It's overkill in those cases where I my JS is so little that it's justified to live inline in Razor pages. So really those cases where I don't use any JS templates anyway.

This move can only be explained by the relevant people not knowing what their are doing.

**Comment by jtheisen** [comment]
@tpetrina Also, Microsoft doesn't support anything anyway. They release admittedly fairly good stuff, but that's it. The only reason I want to use those templates is because it's easy to get react hot loading wrong without it. On Windows, if you have spaces in the path name, as you often do as it contains your username, hot loading silently won't work. MS don't care as they are now run by people who use Linux anyhow.

**Comment by jtheisen** [comment]
Besides, why on earth are you people hiding those templates behind command line tools anyway? What's wrong with just keeping them in a GitHub repo for convenient cloning with variants on different branches? That way users could immediately see what's changed on some minor version bump and check out an earlier version if they want what they got last time they used it.

**Comment by jtheisen** [comment]
[I filed a dedicated ticket with my complaints.](https://github.com/aspnet/Templating/issues/708)
